src/scarabee/_scarabee/cmfd.cpp

```diff
@@ -187,14 +213,33 @@ std::optional<std::size_t> CMFD::get_surface(const Vector& r,
   const double dy_n = std::abs(y_n.y0() - r.y());
   const double dy_p = std::abs(y_p.y0() - r.y());
 
-  if (dx_n < SURFACE_COINCIDENT) return j * x_bounds_.size() + i;
-  if (dx_p < SURFACE_COINCIDENT) return j * x_bounds_.size() + i + 1;
-  if (dy_n < SURFACE_COINCIDENT) return nx_surfs_ + i * y_bounds_.size() + j;
-  if (dy_p < SURFACE_COINCIDENT)
-    return nx_surfs_ + i * y_bounds_.size() + j + 1;
+  // start with corners, then sides
+  // top right
+  if (dx_p < SURFACE_COINCIDENT && dy_p < SURFACE_COINCIDENT) {
+    surface.crossing = CMFDSurfaceCrossing::Type::TR;
+  } else if (dx_p < SURFACE_COINCIDENT && dy_n < SURFACE_COINCIDENT) {
+    surface.crossing = CMFDSurfaceCrossing::Type::BR;
+  } else if (dx_n < SURFACE_COINCIDENT && dy_n < SURFACE_COINCIDENT) {
+    surface.crossing = CMFDSurfaceCrossing::Type::BL;
+  } else if (dx_n < SURFACE_COINCIDENT && dy_p < SURFACE_COINCIDENT) {
+    surface.crossing = CMFDSurfaceCrossing::Type::TL;
+  } else if (dx_p < SURFACE_COINCIDENT) {
+    surface.crossing = CMFDSurfaceCrossing::Type::XP;
+  } else if (dx_n < SURFACE_COINCIDENT) {
+    surface.crossing = CMFDSurfaceCrossing::Type::XN;
+  } else if (dy_p < SURFACE_COINCIDENT) {
+    surface.crossing = CMFDSurfaceCrossing::Type::YP;
+  } else if (dy_n < SURFACE_COINCIDENT) {
+    surface.crossing = CMFDSurfaceCrossing::Type::YN;
+  }
 
-  // If we get here, we aren't on a surface
-  return std::nullopt;
+  else {
```

Comment:
Please move this up to be attached to the closing bracket of the previous else if statement.